sources/packages/web/src/views/aest/institution/ProgramDetails.vue

```diff
@@ -1,29 +1,71 @@
 <template>
   <v-container>
-    <div class="mb-4">
-      <p class="muted-heading-text">
-        <a @click="goBack()">
-          <v-icon left> mdi-arrow-left </v-icon> Back all programs</a
-        >
-      </p>
-      <p class="heading-x-large">View program</p>
-    </div>
+    <header-navigator
+      title="Back all programs"
+      :routeLocation="{
+        name: AESTRoutesConst.INSTITUTION_PROGRAMS,
+        params: { institutionId: institutionId },
+      }"
+      subTitle="View program"
+    >
+      <template #buttons>
+        <div v-if="isPendingProgram">
+          <v-btn
+            outlined
+            :color="COLOR_BLUE"
+            class="mr-2"
+            @click="declineProgram"
+            >Decline</v-btn
+          >
+          <v-btn class="primary-btn-background" @click="approveProgram"
+            >Approve program</v-btn
+          >
+        </div>
+      </template>
+    </header-navigator>
     <ManageProgramAndOfferingSummary
       :programId="programId"
       :locationId="locationId"
-      :clientType="ClientIdType.AEST"
+      :educationProgram="educationProgram"
+      :institutionId="institutionId"
+    />
+    <!-- approve program modal -->
+    <ApproveProgramModal
+      ref="approveProgramModal"
+      @submitData="submitApproveProgram"
```

Comment:
I really don't get why we cannot simply return a Boolean value from modal promise instead of create a event/subscribe/create callback. As always, not a blocker, but IMO the code without the event is more simple.